src/dust_rng.cpp

```diff
@@ -389,6 +389,50 @@ cpp11::sexp dust_rng_multinomial(SEXP ptr, int n,
   return ret;
 }
 
+template <typename real_type, typename T>
+cpp11::sexp dust_rng_hypergeometric(SEXP ptr, int n,
+                                    cpp11::doubles r_n1, cpp11::doubles r_n2,
+                                    cpp11::doubles r_k, int n_threads) {
+  T *rng = cpp11::as_cpp<cpp11::external_pointer<T>>(ptr).get();
+  const int n_streams = rng->size();
+  cpp11::writable::doubles ret = cpp11::writable::doubles(n * n_streams);
+  double * y = REAL(ret);
+  
+  const double * n1 = REAL(r_n1);
+  const double * n2 = REAL(r_n2);
+  const double * k = REAL(r_k);
+  auto n1_vary = check_input_type(r_n1, n, n_streams, "n1");
+  auto n2_vary = check_input_type(r_n2, n, n_streams, "n1");
+  auto k_vary = check_input_type(r_k, n, n_streams, "k");
+
+  dust::utils::openmp_errors errors(n_streams);
+
+#ifdef _OPENMP
+  #pragma omp parallel for schedule(static) num_threads(n_threads)
+#endif
+  for (int i = 0; i < n_streams; ++i) {
```

Comment:
no tests for the case where the `rng` is distributed/`n_streams > 1`?

Reply:
there aren't for most of the distributions tbh, they probably should be added (though this class is really mostly intended for debugging). I've added one in 06c9ec6 though as you're right there _is_ logic here to be tested